docs/readmes/proposals/p019_enodeb_cbrs_support.md

```diff
@@ -0,0 +1,164 @@
+can be obtained from third party spectrum databases called SAS. In Magma this
+requires communication between the enodebd service that manages the radio
+configuration and the domain proxy that runs in the orchestrator that acts as
+a single interface point towards the SAS on behalf of all radios on the network
+
+There are many advantages to having communication go through the proxy as
+opposed to directly from the enodeb including security. Further, having the
+desired state of the network including the radio parameters managed in the
+orchestrator has been a primary goal of Magma.
+
+Goals
+- Allow enodebs to obtain spectrum leases from SAS on bring up
+- Allow enodebs to heartbeat to the SAS and if needed the SAS can revoke the
+  grant
+- Deregister a cbrs eNB from the SAS when it is deleted from magma.
+- Relinquish a grant if the eNB stops radiating.
+
+Non-goals/Simplifications
+- For simplicity we collapse the need for a SAS grant with the transmit enabled
+  status of the radio. i.e. if a CBRS radio is transmitting we assume it needs
+  a grant.
+
+Proposal
+
+The top level goal of this proposal is to have the domain proxy on the
+orchestrator to serve as the truth and act as a desired state store for the
+enodeb and rely on a control channel between enodebd->domain proxy and
+enodeb->enodebd to keep the actual state in sync with the desired state.
+
+User facing:
+- NMS: NMS will explicitly support specific CBRS certified radios from vendors.
+  The knowledge of these radios being CBRS certified is obtained out of band.
+  i.e. for practical purposes the onboarding of these radios are going to be
+  similar to what we have today.
+
+API changes:
+- Swagger: Swagger will introduce a new vendor name for CBRS certified radios
+
+**mconfig changes:**
+- mconfig will be extended to introduce a new boolean field indicating if the
+  radio is a CBRS radio.
+
+**enodebd changes: (only applies to CBRS enodebs)**
+
+There are two subcases for enodeb bring up, eNB provisioned for first time/after
+factory reset. eNB reconnecting to enodebd after reboot/disconnect etc.
+
+**New enodebd provisioning:**
+
+enodebd is in the new enodeb provisioning flow if there is no SAS related
+state for the eNB in Redis.
+
+Towards eNB:
+
+Precondition: SAS state in Redis and in-memory is none.
+- Parse Inform message as today, if serial number maps to enodeb with CBRS
+  enabled and gps location absent, transition to enabling GPS state (new state),
+  else complete session (same complete session as today).
+- Enable GPS using model specific tr-69 node, disable transmit transition to
+  complete session
+- Read GPS location, serial number as part of next inform, complete session.
+- Radio status here is GPS is enabled, but radio is not transmitting
+
+Towards cloud:
+
+Precondition: GPS state, serial number available.
+- Make GRPC call to domain proxy for registration request with the following
+  fields populated.
+  ```
+  “cbsdSerialNumber”: “abcd1234”,
```

Comment:
@xbend to take a pass and add parameters that are missing based on his exploration. Thanks again!